### DISCUSSION TITLE
Apply roll to third column

### DISCUSSION THREAD

**Discussion by Lab9Pro-Service** [ImageMagick/ImageMagick#3226] [Help] [answered]
Hello,

I would like to build a command which takes an input image and makes a pattern out of it. The pattern should have a vertical shift in the second column and all subsequent columns. 

If I use this command:
"C:\Program Files\ImageMagick-7.0.8-Q8\convert.exe" "E:/path/to/square.jpg" -flip ( -clone 0 -roll +0+300 ) +append -write mpr:cell +delete -size 4500x3373 tile:mpr:cell -background "white" -flatten -flip -density 72 -units PixelsPerInch -profile "E:/path/to/square.icc" -quiet "E:/path/to/square_out.jpg"

On this image:
![square](https://user-images.githubusercontent.com/52193770/107665516-ccf92280-6c8d-11eb-89da-223192c15264.jpg)

Then I get this result:
![square_out](https://user-images.githubusercontent.com/52193770/107665612-e7330080-6c8d-11eb-8289-019af0ab0c3b.jpg)

As you can see there is a vertical shift in the second column. However, the pattern in the third column has the same hight as the pattern in the first column. I would like that the pattern in the third column also shifts 300px compared to the second column. Is there some way to do this?

(The two -flip instances is to get the pattern building to start from the lower left instead of the top left)

**Comment by snibgo** [comment]
Many methods are possible. For example, roll the input, then tile it to the full size, then make a crop of the left column, de-roll that, and "-layers flatten" it back.

But your input is a graphic, in CMYK space, with CMYK profile. Do you want that? IM works more easily in RGB colorspaces, eg sRGB.

**Reply by Lab9Pro-Service** [reply]
My graphic is indeed CMYK and this is intended since it will be printed. 

I don't understand why it would make a difference here?

**Comment by GeeMack** [comment, marked as answer]
If I understand the question, you might try adding another clone inside the parentheses, then rolling it another 300 pixels...

    convert square.jpg -flip ^
       ( -clone 0 -roll +0+300 -clone 0 -roll +0+300 ) ^
       +append -write mpr:cell +delete -size 4500x3373 tile:mpr:cell ^
       -background "white" -flatten -flip square_out.jpg

The square is cloned inside the parentheses and rolled down 300 pixels. Then it's cloned again and rolled again. This time the first clone, already rolled 300 pixels, will now be moved 300 more. The three images are appended just as the two images in your command, and it continues from there.

The result is each column of the output tile is shifted one-third of a square more than the previous.

![square_out](https://user-images.githubusercontent.com/65475279/107694692-49333c00-6c75-11eb-9bd3-8c0ecb002fcf.jpg)

**Reply by Lab9Pro-Service** [reply]
Thanks! That's working!
I already tried to duplicate that piece, but I duplicated it including the parentheses, which was not working.
